### PR TITLE
Support ForeachBaggageItem

### PR DESCRIPTION
**Pull request by yurishkuro**
@bensigelman

### CODE REVIEW

File: ext/tags_test.go [hunk 1]
```diff
@@ -40,68 +40,30 @@ func TestPeerTags(t *testing.T) {
...
 
-	rawSpan := span.(*noopSpan)
+	rawSpan := span.(*mocktracer.MockSpan)
 	assertEqual(t, "my-awesome-library", rawSpan.Tags["component"])
 	assertEqual(t, uint16(1), rawSpan.Tags["sampling.priority"])
 	assertEqual(t, true, rawSpan.Tags["error"])
 }
-
-// noopTracer and noopSpan with span tags implemented
-type noopTracer struct{}
```
Comment (bensigelman) [inline ext/tags_test.go]: oh good

File: span.go [hunk 2]
```diff
@@ -87,6 +87,10 @@ type Span interface {
 	// See the `SetBaggageItem` notes about `restrictedKey`.
 	BaggageItem(restrictedKey string) string
 
+	// ForeachBaggageItem allows reading all baggage items stored in the span.
+	// The handler function will be called for each baggage key/value pair.
+	ForeachBaggageItem(handler func(k, v string))
```
Comment (bensigelman) [inline span.go]: sometimes these sorts of handler functions return a bool to indicate that the internal iterator can/should stop... thoughts?

**Comment by bensigelman** [comment]
Basically LGTM. Would you be up for sending out the trivial change to support this in `basictracer-go` just to keep it up to date? If you don't have time I can do it.

File: span.go [hunk 3]
```diff
@@ -87,6 +87,14 @@ type Span interface {
 	// See the `SetBaggageItem` notes about `restrictedKey`.
 	BaggageItem(restrictedKey string) string
 
+	// ForeachBaggageItem allows reading all baggage items stored in the span.
```
Comment (dominikh) [inline span.go]: Is it worth documenting whether ForeachBaggageItem has a stable iteration order, or if the order can differ between consecutive calls of ForeachBaggageItem?

Comment (yurishkuro) [inline span.go]: @dominikh https://github.com/opentracing/opentracing-go/commit/9e6d3e21876169b028e75016a19d75c4ad54cb11

**Comment by bensigelman** [comment]
LGTM, thanks.

**Pull request merged**